src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -115,6 +116,13 @@ impl ScalarFunc {
         digest.write(&d);
         Ok(Some(i64::from(digest.sum32())))
     }
+
+    #[inline]
+    pub fn pow(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
+        let x = try_opt!(self.children[0].eval_real(ctx, row));
+        let y = try_opt!(self.children[1].eval_real(ctx, row));
+        Ok(Some(x.pow(y)))
```

Comment:
Should we check `is_infinite` like tidb did? https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L1059 @zz-jason

Reply:
```rust
    #[inline]
    pub fn pow(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
        let x = try_opt!(self.children[0].eval_real(ctx, row));
        let y = try_opt!(self.children[1].eval_real(ctx, row));
        let pow = x.pow(y);
        if pow.is_infinite() || pow.is_nan() {
           return Ok(Some(0.0));
        }
        Ok(Some(x.pow(y)))
    }
```
If it is infinite or NAN return 0?